lib/AST/Type.cpp

```diff
@@ -3240,12 +3240,29 @@ TypeSubstitutionMap TypeBase::getContextSubstitutions(const DeclContext *dc) {
 
 SubstitutionMap TypeBase::getContextSubstitutionMap(
   ModuleDecl *module, const DeclContext *dc) {
+  auto lookup = LookUpConformanceInModule(module);
+
+  // If we're substituting in a protocol, we can avoid the full
+  // generic-signature-based lookup, because the substitutions are easy to
+  // compute. It is important to do this, because name-lookup needs to use this,
+  // and the generic signature of a protocol may not yet be computed.
```

Comment:
I'm not sure it is trivial enough: don't all the implied requirements have to be stored too (in the `GenericSignatureBuilder`)?